### DISCUSSION TITLE
Problem to connect the Lighthouse Basestations

### DISCUSSION THREAD

**Discussion by omorovi** [bitcraze/discussions#1463] [Q&A]
I am writing to seek your assistance regarding an issue I am encountering with the Lighthouse positioning system. Specifically, I am having difficulty connecting multiple Base Stations. The problem I am facing is as follows: After successfully connecting one Base Station in the client software, I am unable to detect and connect another Base Station during the scanning process. I would greatly appreciate your guidance in resolving this issue.

**Comment by evoggy** [comment]
Hi!

Do you mean when you are connecting to the base stations using USB for configuring the channel? Does it work if you restart the client? This procedure only has to be done once for setting up the system, so one workaround could be to restart the client in between connecting each base station.

**Reply by omorovi** [reply]
Yes I mean the connection using USB. After connecting both of my 2 Basestation I can get data form only one base station, that means one of my base station is not connected. Here is the snippet,

![Basestation data](https://github.com/user-attachments/assets/a66c9377-9f00-4309-962d-9d0d531be710)

I have restart the client to connect another Basestation, but I did not get any result. Could you please suggest how can I get data for both of the BaseStations?

**Reply by evoggy** [reply]
Ok, can you double check that you changed the address of the second base station to 2?

**Reply by omorovi** [reply]
Yes, I have checked this several time. But I did not get any changes. Is there any updated "Client" software availabe?

**Reply by evoggy** [reply]
Please have a look at the [release page](https://github.com/bitcraze/crazyflie-clients-python/releases), it's always good to have the latest client. But from what I can see there hasn't been any changes here.

So if you turn off the base station you have assigned id 1 to you get no data at all? Can you describe the setup a bit more? Can the Crazyflie see both of the base stations at once? How far away are they? Is there a white or green LED lit up on the front of the base stations? Are they Lighthouse V1 or V2 base stations?

**Reply by omorovi** [reply]
So, if you turn off the base station you have assigned id 1 to you get no data at all? 
Yes, first, I turned off one base station and got the Base station Status. And then turned off the Base station 2 and saw that same data in the Base station status (only on column 1 of the table). That means both base stations are connected but I do not get data of geometry and estimator data. Then what could be the problem? 
Can the Crazyflie see both of the base stations at once? Yes, the drone can see both base stations at the same time. The base stations are mounted 1-1.5-meter distance in the wall. The base stations are V2, and the LED is green first and then white. 

I understand that both base stations are connected but I do not get data of geometry and estimator data. Then what could be the problem? 

![WhatsApp Image 2024-07-19 at 10 44 09 AM](https://github.com/user-attachments/assets/3deeb778-7030-4110-9237-01b2f08ed27b)
![WhatsApp Image 2024-07-19 at 10 44 15 AM](https://github.com/user-attachments/assets/ce9c54df-24e3-4ab1-afa0-99318888dfe6)

**Reply by knmcguire** [reply]
Hi! 

First of all, you'll need to remove the film on the basestations. It won't work as well with it.

Secondly, if you tried to turn off basestation 2 and turned on one, and only column 1 showed data, and did the same with turning on basestation  2 and turning off basestation 1, then I'm afraid that still something went wrong with the assigning of the IDs as they are currently set to 1, unfortunately there is no other explanation.

The geometry and estimation data will only turn green once you have done the geometry estimation and able to receive the calibration variables from the basestation. That is currently not possible if both basestations are set to 1.

Perhaps it would be good to start fresh from the lighthouse getting started guide, which also have a video guidance: https://www.bitcraze.io/documentation/tutorials/getting-started-with-lighthouse/

**Comment by omorovi** [comment]
Is there any option to reset the lighthouse basestations?

**Reply by knmcguire** [reply]
There probably is. These are the official SteamVR basestations so you can probably see that on the forum: https://steamcommunity.com/app/250820/discussions/0/1751275054055546812/

These instructions will work as well: https://www.vive.com/us/support/vive/category_howto/updating-the-base-stations-via-cable-connection.html#:~:text=The%20hardware%20may%20be%20purchased%20separately%20from%20the,their%20power%20adapters%2C%20and%20then%20carefully%20unmount%20them.

Don't forget to also remove the film on the basestations. That white arrow shouldn't be there and the plastic is blocking the signals

**Comment by omorovi** [comment]
Earlier the problem maybe I made that both of the basestation have assined in a same channel. If I can reset the channel so that I can set it again. let me know if there is any option with the basestation v2 to reset.

But we have also another problem, I have started with fresh installation but the problem I am facing that I am not sure that the basestation are getting assigned because there is no sing or pop up showing success! Please suggest me with that. 

![image](https://github.com/user-attachments/assets/94c86d76-d533-4872-a7df-5f040c777b81)

**Reply by knmcguire** [reply]
It seems that it is able to find it but it is perhaps not able or allowed to to read from it.

This is how it looks on my computer (Windows11, already preconfigured, but you should be at least able to see something)

![image](https://github.com/user-attachments/assets/5c06ef70-8ce7-4fec-9537-c2ff1cefaded)


In the terminal in which you started the cfclient in, did any message or error occure when you tried scan or write the id to the basestations?

**Reply by omorovi** [reply]
I did not get any message or error in the cfclient software. I have tried in 2 computer, one is Ubuntu 22.04 jammy and another one is windows 10. I did not get any error or any message. The Basestation configuration window is same. This is why I got confisued and maybe overlap the basestation in one channel. Now how can I solve this problem? is there any way for me. Please suggest me. Thanks

**Reply by knmcguire** [reply]
Another thing that might be wrong is that the basetstation is not powered on. It should have 2 cables inserted, the power cable and the microusb for the computer. 

When you try to communicate with the basestation with the dialog, is the LED on?

**Comment by omorovi** [comment]
Yes the basestations have been powered and assigned with the MicroUSB.
The LED is on and it's green. Thanks 

![WhatsApp Image 2024-08-02 at 3 38 33 PM](https://github.com/user-attachments/assets/70a765bd-771f-46a4-933a-57a2cc263138)

**Reply by knmcguire** [reply]
Alright, let's try to debug it further. I'm assuming that you are also only connecting only base station at a time to the computer right?

- On ubuntu, could you give me the output of `lsusb ` in a terminal with one of these connected.
- On Windows, can you see the connected base station in the 'Device manager'

**Comment by omorovi** [comment]
Here is the list of usb connection in Ubuntu after connecting only one Basestation:
![image](https://github.com/user-attachments/assets/db0b8a4b-84a0-4bb8-a0f9-8d20d9ef960e)


Here is the list of USB connected in my window computer after connecting one BaseStation:
![image](https://github.com/user-attachments/assets/556de7e3-cdec-41aa-8b6d-4a30c760187f)

**Reply by knmcguire** [reply]
So strange! All same for me. Let's debug further:

On Ubuntu, connect and reconnect the basestation usb cable and check with `dmesg` which ACM port it is connected to. then do `ls -l /dev/ttyACM0` if you have the correct rights. For me it says: 
```
crw-rw---- 1 root dialout 166, 0 Aug  7 11:23 /dev/ttyACM0
```

Other than that I'm not sure what would help out then other than to send the basestations to our office and do the configuring for you... Another thing you can try is to change the channels via steamVR/Vives instructions here: https://www.vive.com/nz/support/vive-pro/category_howto/configuring-the-base-station-channels.html.
